pkg/pq/priority_queue.go

```diff
@@ -51,16 +51,31 @@ func (pq *PriorityQueue) Len() int {
 }
 
 func (pq *PriorityQueue) Release(value Value) {
-	queue := &internalQueue{}
-	heap.Init(queue)
+	idx := pq.findIdx(func(item *Item) bool {
+		return item.value == value
+	})
 
-	for _, item := range *pq.queue {
-		if item.value != value {
-			heap.Push(queue, item)
-		}
+	if idx == -1 {
+		return
+	}
+
+	lastItem := pq.queue.Pop().(*Item)
+
+	if (*pq.queue).Len() == 0 || idx == (*pq.queue).Len() {
```

Comment:
Since we can execute `Len()` without dereferencing, why not express it simply?
```suggestion
	if pq.queue.Len() == 0 || idx == pq.queue.Len() {
```